### ISSUE TITLE
2.8.0 introduces breaking change

### ISSUE THREAD

**Issue by nickesk** [fluttercandies/flutter_photo_manager#1034]
### Version

2.8.0

### Platforms

Android

### Device Model

all

### flutter info

```sh
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.9, on macOS 13.0.1 22A400 darwin-arm64,
    locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version
    33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.84.1)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!
```

### How to reproduce?

set `compileSdkVersion 33` in a flutter project and try to build with `photo_manager` version `2.8.0` or greater.

**Comment by nickesk** [comment]
As this was a minor version release, a breaking change must not be introduced per the semver spec.

**Comment by CaiJingLong** [comment]
First, photo_manager does not fully follow the semver spec, we only upgrade major versions when code incompatibilities are introduced. The breaking change in 2.8 is only a change in API behavior, there is no incompatibility in the code.

Second, if your usage follows the README, the code does **not need to be modified**.

**Comment by harrynguyen2510** [comment]
There is a bug: **Could not build the precompiled application for the device.
Error (Xcode): ../../../.pub-cache/hosted/pub.dev/photo_manager-2.8.1/lib/src/internal/image_provider.dart:71:5: Error: Type 'DecoderCallback' not found.**

**Comment by chopper985** [comment]
> First, photo_manager does not fully follow the semver spec, we only upgrade major versions when code incompatibilities are introduced. The breaking change in 2.8 is only a change in API behavior, there is no incompatibility in the code.
> 
> Second, if your usage follows the README, the code does **not need to be modified**.
> 
> By the way, Flutter SDK also has breaking changes in minor version, such as 3.16.0 remove some fields and methods with `ImageProvider`.

When will this bug be fixed for version 3.16.0 of Flutter
